stan/math/prim/core/init_threadpool_tbb.hpp

```diff
@@ -84,13 +84,17 @@ inline int get_num_threads() {
  * @throws std::runtime_error if the value of STAN_NUM_THREADS env. variable
  * is invalid
  */
-inline tbb::global_control& init_threadpool_tbb() {
+inline tbb::task_arena& init_threadpool_tbb() {
+  auto tbb_gc_mp = tbb::global_control::max_allowed_parallelism;
   int tbb_max_threads = internal::get_num_threads();
 
-  static tbb::global_control tbb_scheduler(
-      tbb::global_control::max_allowed_parallelism, tbb_max_threads);
+  tbb::global_control tbb_gc(tbb_gc_mp, tbb_max_threads);
 
-  return tbb_scheduler;
+  static tbb::task_arena tbb_arena(tbb_max_threads, 1,
+                                   tbb::task_arena::priority::normal);
+  tbb_arena.initialize();
+
+  return tbb_arena;
```

Comment:
If this function aims to initialize the Intel TBB threadpool and global scheduler, it should return an initialized `tbb_arena` and not a `global_control` object. The old interface returns a `task_scheduler_init` object, which is now removed in the new TBB interface.